src/elements/components/TextNodes.tsx

```diff
@@ -82,56 +84,87 @@ function TextNodes({
       });
     } else if (actions.length > 0) delta = applyNewDelta(delta);
 
+    // If text_mode property is set to 'data', then we don't want to render the text_formatted
+    // property, instead we the text from the data element specified in the text_source property
+    let textFromData = null;
```

Comment:
lines 89 - 93 seem like they could be moved inside of the if statement on line 95